odoo/http.py

```diff
@@ -995,6 +995,63 @@ def delete_from_identifiers(self, identifiers):
             with contextlib.suppress(OSError):
                 os.unlink(fn)
 
+    @contextlib.contextmanager
```

Comment:
Everything I say below is debatable, I am not sure myself what we should do. The only thing I know is that we should keep it KISS as much as possible.

I have mixed feelings regarding the fact to make a tool out of this / put this in odoo/http, in the class `FilesystemSessionStore`.

Making it a tool makes sense when it's used at more than one place. Do we expect we will need this mechanism elsewhere than in the reports ? I do not say it will never be the case, it might very well be, but currently this is the only place we know which requires this and I do not foresee amother case. Making it a tooling seems prematured optimization to me.

In addition I do not think I would put this directly in the `FilesystemSessionStore` class. It seems to me out of scope of that class.

To keep it stupid simple, I would simply put everything in `odoo/addons/base/models/ir_actions_report.py`, directly in the `_run_wkhtmltopdf` method without using any factorization method, or maybe a in standalone method / out of the model class. There are already a few anyway in this file: `_get_wkhtmltopdf_bin`, `_get_wkhtmltoimage_bin`, `_split_table`.

However, to keep it elegant in addition to keeping it simple, I agree it would be better to abstract out the session_store / session save/delete mechanism out of this file. But currently its not the case fully either as it still need to import `odoo.http.root` within `ir_actions_report.py`. So maybe we should focus on keeping it simple for the moment rather than elegant,

Currently the decorator is on the whole method `_run_wkhtmltopdf` while the new/duplicate session is actually needed in only 2 short parts of the code:
- when the cookie jar is generated, which must use the sid of the new duplicated session,
- when the call to the actual wkhtmtopdf executable is done>

With the decorator on the whole method, it makes harder for a developer to understand in which part of the method this decorator is actually required, On which part of the code the decorator applies. With a `with temporary_session(...)` rather than a decrator on the method, it would already scope it better. But it's not so easy to put either as the 2 code parts mentioned above are not grouped together.

And adding a decorator for a lone use case is also not very KISS.

I think I would simply duplicate the session within the `if request and request.db and request.session.sid:` block of `_run_wkhtmltopdf`, not even factorizing in a standalone method.
And then delete the session within a `finally` of the already existing `try/except` block wrapping the popen call to the wkhtmltopdf binary, to make sure it's well deleted even in case of exception.

Regarding the unit test, currently put in `test_http`: it tests the behavior of the tool `temporary_session`, which we will no longer have if you apply my above comment.

However we can do a unit test more based on the report generation rather than the behavior of the `temporary_session` tool: generate a report, and assert that no new device log is created.

You might actually end up with less diff, as you night no longer need the changes in `url_open`, adding that `cookies` parameter.